### ISSUE TITLE
feat: add the optional argument `cache` to the constructor

### ISSUE THREAD

**Issue by tmorin** [orbitdb/orbit-db-keystore#41]
Hi,

Could it be possible to provide the instance of the cache to use as an argument of the constructor as below?

```typescript
const keystore1 = new Keystore()
const keystore2 = new Keystore(undefined, new LRU(10))
const keystore3 = new Keystore(level('./somewhere'), new LRU(10))
const keystore4 = new Keystore(level('./somewhere'))
```

Or using an options object?

```typescript
const keystore1 = new Keystore()
const keystore2 = new Keystore({
  cache: new LRU(10)
})
const keystore3 = new Keystore({
  store: level('./somewhere'),
  cache: new LRU(10)
})
const keystore4 = new Keystore({
  store: level('./somewhere')
})
// for compatibilities
const keystore4bis = new Keystore(level('./somewhere'))
```

If you are agree I can do the pull request with the additional tests.

Best regards

**Comment by aphelionz** [comment]
What do you think @shamb0t and @haadcode

**Comment by shamb0t** [comment]
This would be a nice addition, thank you @tmorin! Though its a breaking change I would prefer the options object approach to avoid passing `undefined` in the args, what do you think?

**Comment by tmorin** [comment]
hi @shamb0t , @aphelionz 
The options' object is my preference too.
Best regards

**Comment by shamb0t** [comment]
closed by #42, thanks @tmorin!
